src/components/character-controller.js

```diff
@@ -0,0 +1,300 @@
+      -this.data.snapRotationRadian
+    );
+  },
+
+  play: function() {
+    var eventSrc = this.el.sceneEl;
+    eventSrc.addEventListener("stop_moving", this.onStopMoving);
+    eventSrc.addEventListener("translateX", this.onTranslateX);
+    eventSrc.addEventListener("translateY", this.onTranslateY);
+    eventSrc.addEventListener("translateZ", this.onTranslateZ);
+    eventSrc.addEventListener("action_move_forward", this.onMoveForward);
+    eventSrc.addEventListener(
+      "action_dont_move_forward",
+      this.onDontMoveForward
+    );
+    eventSrc.addEventListener("action_move_backward", this.onMoveBackward);
+    eventSrc.addEventListener(
+      "action_dont_move_backward",
+      this.onDontMoveBackward
+    );
+    eventSrc.addEventListener("action_move_left", this.onMoveLeft);
+    eventSrc.addEventListener("action_dont_move_left", this.onDontMoveLeft);
+    eventSrc.addEventListener("action_move_right", this.onMoveRight);
+    eventSrc.addEventListener("action_dont_move_right", this.onDontMoveRight);
+
+    eventSrc.addEventListener("rotateY", this.onRotateY);
+    eventSrc.addEventListener("action_snap_rotate_left", this.onSnapRotateLeft);
+    eventSrc.addEventListener(
+      "action_snap_rotate_right",
+      this.onSnapRotateRight
+    );
+    eventSrc.addEventListener("boost", this.onBoost);
+  },
+
+  pause: function() {
+    var eventSrc = this.el.sceneEl;
+    eventSrc.removeEventListener("stop_moving", this.onStopMoving);
+    eventSrc.removeEventListener("translateX", this.onTranslateX);
+    eventSrc.removeEventListener("translateY", this.onTranslateY);
+    eventSrc.removeEventListener("translateZ", this.onTranslateZ);
+    eventSrc.removeEventListener("action_move_forward", this.onMoveForward);
+    eventSrc.removeEventListener(
+      "action_dont_move_forward",
+      this.onDontMoveForward
+    );
+    eventSrc.removeEventListener("action_move_backward", this.onMoveBackward);
+    eventSrc.removeEventListener(
+      "action_dont_move_backward",
+      this.onDontMoveBackward
+    );
+    eventSrc.removeEventListener("action_move_left", this.onMoveLeft);
+    eventSrc.removeEventListener("action_dont_move_left", this.onDontMoveLeft);
+    eventSrc.removeEventListener("action_move_right", this.onMoveRight);
+    eventSrc.removeEventListener(
+      "action_dont_move_right",
+      this.onDontMoveRight
+    );
+    eventSrc.removeEventListener("rotateY", this.onRotateY);
+    eventSrc.removeEventListener(
+      "action_snap_rotate_left",
+      this.onSnapRotateLeft
+    );
+    eventSrc.removeEventListener(
+      "action_snap_rotate_right",
+      this.onSnapRotateRight
+    );
+  },
+
+  onStopMoving: function(event) {
+    this.accelerationInput.set(0, 0, 0);
+  },
+
+  onTranslateX: function(event) {
+    // The type check here is because the last axismove event that is captured here sends
+    // the el as the event.detail. This should probably be caught earlier.
```

Comment:
Is this still true? I can't tell where the event.detail is set to el.

Reply:
Not true anymore. Thanks for catching.